compressor_test.go

```diff
@@ -36,3 +40,63 @@ func TestSnappyCompressor(t *testing.T) {
 		t.Fatal("failed to match the expected decoded value with the result decoded value.")
 	}
 }
+
+func TestBlobCompressor(t *testing.T) {
+	session := createSession(t)
+	defer session.Close()
+	// TypeVarchar, TypeAscii, TypeBlob, TypeText
+	if err := createTable(session, `CREATE TABLE gocql_test.test_blob_compressor (
+			testuuid       timeuuid PRIMARY KEY,
+			testvarchar    varchar,
+			testblob       blob,
+			testascii      ascii,
+			testtext 	   text,
+		)`); err != nil {
+		t.Fatal("create table:", err)
+	}
+	m := make(map[string]interface{})
+
+	BlobCompressor = lz4.NewBlobCompressor(100)
+
+	originalBlob := strings.Repeat("1234567890", 20)
+
+	m["testuuid"] = TimeUUID()
+	m["testvarchar"] = originalBlob
+	m["testblob"] = []byte(originalBlob)
+	m["testascii"] = originalBlob
+	m["testtext"] = originalBlob
+	sliceMap := []map[string]interface{}{m}
+	if err := session.Query(`INSERT INTO gocql_test.test_blob_compressor (testuuid, testvarchar, testblob, testascii, testtext) VALUES (?, ?, ?, ?, ?)`,
+		m["testuuid"], m["testvarchar"], m["testblob"], m["testascii"], m["testtext"]).Exec(); err != nil {
+		t.Fatal("insert:", err)
+	}
+	if returned, retErr := session.Query(`SELECT * FROM test_blob_compressor`).Iter().SliceMap(); retErr != nil {
+		t.Fatal("select:", retErr)
+	} else {
+		if diff := cmp.Diff(sliceMap, returned); diff != "" {
+			t.Fatal("mismatch in returned map", diff)
+		}
+	}
+
+	// Test for Iter.MapScan()
+	{
+		testMap := make(map[string]interface{})
+		if !session.Query(`SELECT * FROM test_blob_compressor`).Iter().MapScan(testMap) {
+			t.Fatal("MapScan failed to work with one row")
+		}
+		if diff := cmp.Diff(sliceMap[0], testMap); diff != "" {
+			t.Fatal("mismatch in returned map", diff)
+		}
+	}
+
+	// Test for Query.MapScan()
+	{
+		testMap := make(map[string]interface{})
+		if session.Query(`SELECT * FROM test_blob_compressor`).MapScan(testMap) != nil {
+			t.Fatal("MapScan failed to work with one row")
+		}
+		if diff := cmp.Diff(sliceMap[0], testMap); diff != "" {
+			t.Fatal("mismatch in returned map", diff)
+		}
+	}
```

Comment:
IIUC this test checks identity of insert / select, so it would pass if there was no compressor set / marshal module didn't use the compressor, right?
If so, then it should be improved - it's an integration test so it should check that the new feature works properly with the rest of the driver, right now it doesn't do this.